### PR TITLE
chore: use code linting

### PR DESCRIPTION
**Pull request by mrded**
This change enables a code linting check in the Circle CI.

### CODE REVIEW

**Review by gablaxian — APPROVED** [review]

**Review by carissadean — APPROVED** [review]

File: src/rulesets/properties.ts [hunk 1]
```diff
@@ -176,7 +172,7 @@ export const rules = {
         if (specItem.type === "object") {
           if (!specItem.properties)
             expect.fail("object schema is missing 'properties'");
-          if (specItem.hasOwnProperty("items"))
+          if (Object.prototype.hasOwnProperty.call(specItem, "items"))
```
Comment (mrded) [inline src/rulesets/properties.ts:175]: Linter didn't like it:
```
  175:24  error  Do not access Object.prototype method 'hasOwnProperty' from target object  no-prototype-builtins
```

File: src/rulesets/tests/__snapshots__/specification.test.ts.snap [hunk 2]
```diff
@@ -1,872 +0,0 @@
-// Jest Snapshot v1, https://goo.gl/fbAQLP
-
-exports[`failing case 1: adding a camelCase when pascalCase is required 1`] = `
```
Comment (cmars) [inline src/rulesets/tests/__snapshots__/specification.test.ts.snap:3]: The fact a snapshot test changed as a result of satisfying eslint changes is setting off alarm bells. We have done something to change the behavior of the rules that we may not understand.

I'd prefer to preserve existing behavior and make eslint exceptions until we have actual unit tests (not just snapshot coverage) to feel confident in making these changes.

**Review by cmars — CHANGES_REQUESTED** [review]
I'm +1 on introducing eslint (it's already highlighting some gaps below), but if snapshots change as a result ... I fear we've introduced some subtle change to behavior that we may not fully comprehend or have test coverage for. In those particular cases I'd rather ignore eslint until we have proper test coverage.

FWIW Optic is aware of test coverage gaps and will be introducing a better DSL that should improve coverage and rule-writing ergonomics considerably.

CC @acunniffe

File: src/rulesets/properties.ts (hunk 1 above)
Comment (mrded) [reply src/rulesets/properties.ts:175]: I've checked the tests, and they are not touching the condition. Test coverage is not good here, it's a bit worrying.

However I'm confident on the JS change: 
image

I would suggest making a test case that touches the condition, that will be the best.

File: src/rulesets/properties.ts [hunk 3]
```diff
@@ -4,10 +4,6 @@ import { OpenAPIV3 } from "@useoptic/api-checks";
 import { links } from "../docs";
 import { getBodyPropertyName } from "../names";
 
-const oas3Formats = ["date", "date-time", "password", "byte", "binary"];
-
-const allowedFormats = Array.prototype.concat(oas3Formats, ["uuid"]);
-
```
[the remark this answers is not in the document]
Comment (mrded) [reply src/rulesets/properties.ts:10]: How about I export `allowedFormats`? That will keep the formats and make the linter happy.

[the remark this answers is not in the document]
Comment (mrded) [reply src/rulesets/properties.ts:10]:
```suggestion
export const allowedFormats = Array.prototype.concat(oas3Formats, ["uuid"]);
```

File: src/rulesets/tests/__snapshots__/specification.test.ts.snap (hunk 2 above)
Comment (mrded) [reply src/rulesets/tests/__snapshots__/specification.test.ts.snap:3]: the snapshot has not changed, it was abandoned. We no longer have `specification.test.ts` file and a snapshot. That's why.

File: src/rulesets/properties.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (mrded) [reply src/rulesets/properties.ts:10]: nit: `Array.prototype.concat(oas3Formats, ["uuid"])`  -> `[...oas3Formats, "uuid"]`.

[the remark this answers is not in the document]
Comment (mrded) [reply src/rulesets/properties.ts]: - [x] Keep these in and add an eslint ignore for now
- [x] Add a TODO comment to improve rules to actually check these on string types
- [ ] Open an issue to resolve this gap

**Review by cmars — APPROVED** [review]

File: src/rulesets/properties.ts (hunk 1 above)
Comment (mrded) [reply src/rulesets/properties.ts:182]: I made an issue to add a test coverage here https://github.com/snyk/sweater-comb/issues/236

File: src/rulesets/properties.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (cmars) [reply src/rulesets/properties.ts]: +💯 on the export, happy with how this landed, thanks for this!

**Pull request merged**